test/test_ops.py

```diff
@@ -246,7 +246,10 @@ def test_variant_consistency_eager(self, device, dtype, op):
     # TODO WARNING: inplace x {traced, scripted} not currently tested
     @ops(op_db)
     def test_variant_consistency_jit(self, device, dtype, op):
-        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        test_backward = (
+            (dtype.is_complex and op.test_complex_grad) or
+            (dtype.is_floating_point and (not op.skip_bfloat16_grad or dtype != torch.bfloat16)))
+        samples = op.sample_inputs(device, dtype, requires_grad=test_backward)
```

Comment:
`std` and `var` aren't whitelisted for complex autograd, so `test_complex_grad` needs to never call `backward()` at all. Why wasn't this here already? Well it looks like `sigmoid` is the only other user of `test_complex_grad` and it explicitly skips these two jit tests:

https://github.com/pytorch/pytorch/blob/7fdc6a27b81d1ece894005db3b03860afd935a12/torch/testing/_internal/common_methods_invocations.py#L1608-L1612

Reply:
These changes to the JIT testing look good to me, it was a bad oversight. Thanks for catching it